lib/url.js

```diff
@@ -526,7 +526,12 @@ function getHostname(self, rest, hostname) {
     // Invalid host character
     if (!isValid) {
       self.hostname = hostname.slice(0, i);
-      return `/${hostname.slice(i)}${rest}`;
+      const leftover = hostname.slice(i);
+      // If leftover starts with :, then it represents an invalid port.
+      if (leftover.startsWith(':')) {
```

Comment:
I believe checking this before line 528 would be faster for the invalid route. Something like:

`if (hostname.charCodeAt(i) === 58)`

Reply:
Works for me. Done.